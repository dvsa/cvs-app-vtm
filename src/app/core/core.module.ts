import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { GlobalErrorComponent } from './components/global-error/global-error.component';
import { SpinnerComponent } from './components/spinner/spinner.component';
import { FooterComponent } from './components/footer/footer.component';
import { HeaderComponent } from './components/header/header.component';
import { RouterModule } from '@angular/router';
import { BreadcrumbsComponent } from './components/breadcrumbs/breadcrumbs.component';
import { PageNotFoundComponent } from './components/page-not-found/page-not-found.component';
import { ServerErrorComponent } from './components/server-error/server-error.component';

@NgModule({
  declarations: [FooterComponent, HeaderComponent, GlobalErrorComponent, SpinnerComponent, BreadcrumbsComponent, PageNotFoundComponent, ServerErrorComponent],
  imports: [CommonModule, RouterModule],
  exports: [FooterComponent, HeaderComponent, GlobalErrorComponent, SpinnerComponent, BreadcrumbsComponent]
})
export class CoreModule {}
