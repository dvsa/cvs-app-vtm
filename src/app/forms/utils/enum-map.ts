import { MultiOptions } from '@forms/models/options.model';

export function getOptionsFromEnum(object: object): MultiOptions {
  return Object.values(object).map(value => ({ value, label: value.charAt(0).toUpperCase() + value.slice(1) }));
}

export function getOptionsFromEnumOneChar(object: object): MultiOptions {
  return Object.values(object).map(value => ({ value, label: value.charAt(0).toUpperCase() }));
}

export function getOptionsFromEnumAcronym(object: object): MultiOptions {
  return Object.values(object).map(value => ({ value, label: value.toUpperCase() }));
}
