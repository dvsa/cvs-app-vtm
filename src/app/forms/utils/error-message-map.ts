import { formatDate } from '@angular/common';
import { AsyncValidatorNames } from '@forms/models/async-validators.enum';
import { ValidatorNames } from '@forms/models/validators.enum';

const DEFAULT_LABEL = 'This field';
export const ErrorMessageMap: Record<string, (...args: any) => string> = {
  // Date errors
  invalidDate: (err: { error: boolean; reason: string; index: number }) => `${err.reason}`,
  invalidOption: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is invalid`,
  invalidTestResult: (err: { message: string }) => err.message,

  [ValidatorNames.AheadOfDate]: (err: { sibling: string; date: Date }, label?: string) =>
    `${label || 'This date'} must be ahead of ${err.sibling || 'the previous date'}${err.date ? formatDate(err.date, ' (dd/MM/yyyy)', 'en') : ''}`,
  [ValidatorNames.CustomPattern]: (err: { message: string }, label?: string) => `${label || DEFAULT_LABEL} ${err.message}`,
  [ValidatorNames.DateNotExceed]: (err: { sibling: string; months: number }, label?: string) =>
    `${label || 'This date'} must be less than ${err.months || 'X'} months after ${err.sibling || 'the previous date'}`,
  [ValidatorNames.Defined]: (err: boolean, label?: string) => `${label} is required`,
  [ValidatorNames.FutureDate]: (err: boolean, label?: string) => `${label || 'This date'} must be in the future`,
  [ValidatorNames.Max]: (err: { max: number }, label?: string) => `${label || DEFAULT_LABEL} must be less than or equal to ${err.max}`,
  [ValidatorNames.MaxLength]: (err: { requiredLength: number }, label?: string) =>
    `${label || DEFAULT_LABEL} must be less than or equal to ${err.requiredLength} characters`,
  [ValidatorNames.Min]: (err: { min: number }, label?: string) => `${label || DEFAULT_LABEL} must be greater than or equal to ${err.min}`,
  [ValidatorNames.MinLength]: (err: { requiredLength: number }, label?: string) =>
    `${label || DEFAULT_LABEL} must be greater than or equal to ${err.requiredLength} characters`,
  [ValidatorNames.NotZNumber]: () => "The VRM/Trailer ID cannot be in a format that is 7 digits followed by the character 'Z'",
  [ValidatorNames.PastDate]: (err: boolean, label?: string) => `${label || 'This date'} must be in the past`,
  [ValidatorNames.Pattern]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} must match a pattern`,
  [ValidatorNames.Required]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`,
  [ValidatorNames.RequiredIfEquals]: (err: { sibling: string }, label?: string) => `${label || DEFAULT_LABEL} is required with ${err.sibling}`,
  [ValidatorNames.ValidateDefectNotes]: () => 'Notes is required',
  [ValidatorNames.ValidateProhibitionIssued]: () => 'Prohibition notice has not been issued.',

  [AsyncValidatorNames.RequiredIfNotAbandoned]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`,
  [AsyncValidatorNames.RequiredIfNotFail]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`,
  [AsyncValidatorNames.RequiredIfNotResult]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`,
  [AsyncValidatorNames.RequiredIfNotResultAndSiblingEquals]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`
};
