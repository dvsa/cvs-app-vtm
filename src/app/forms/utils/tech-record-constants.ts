import { FormNode } from '@forms/services/dynamic-form.types';
import { CarTechRecord } from '@forms/templates/car/car-tech-record.template';
import { ApplicantDetails } from '@forms/templates/general/applicant-details.template';
import { HgvAndTrlTypeApprovalTemplate } from '@forms/templates/general/approval-type.template';
import { Audit } from '@forms/templates/general/audit.template';
import { DocumentsTemplate } from '@forms/templates/general/documents.template';
import { HgvAndTrlBodyTemplate } from '@forms/templates/general/hgv-trl-body.template';
import { LettersTemplate } from '@forms/templates/general/letters.template';
import { ManufacturerTemplate } from '@forms/templates/general/manufacturer.template';
import { NotesTemplate } from '@forms/templates/general/notes.template';
import { PlatesTemplate } from '@forms/templates/general/plates.template';
import { HgvDimensionsTemplate } from '@forms/templates/hgv/hgv-dimensions.template';
import { HgvTechRecord } from '@forms/templates/hgv/hgv-tech-record.template';
import { tyresTemplateHgv } from '@forms/templates/hgv/hgv-tyres.template';
import { HgvWeight } from '@forms/templates/hgv/hgv-weight.template';
import { LgvTechRecord } from '@forms/templates/lgv/lgv-tech-record.template';
import { MotorcycleTechRecord } from '@forms/templates/motorcycle/motorcycle-tech-record.template';
import { PsvTypeApprovalTemplate } from '@forms/templates/psv/psv-approval-type.template';
import { PsvBodyTemplate } from '@forms/templates/psv/psv-body.template';
import { PsvBrakesTemplate } from '@forms/templates/psv/psv-brakes.template';
import { PsvDdaTemplate } from '@forms/templates/psv/psv-dda.template';
import { PsvDimensionsTemplate } from '@forms/templates/psv/psv-dimensions.template';
import { PsvNotes } from '@forms/templates/psv/psv-notes.template';
import { PsvTechRecord } from '@forms/templates/psv/psv-tech-record.template';
import { PsvTyresTemplate } from '@forms/templates/psv/psv-tyres.template';
import { PsvWeightsTemplate } from '@forms/templates/psv/psv-weight.template';
import { SmallTrailerTechRecord } from '@forms/templates/small-trailer/small-trailer-tech-record.template';

import { TechRecordReasonForCreationSection } from '@forms/templates/test-records/section-templates/techRecord_reasonForCreation/techRecordReasonForCreation.template';
import { TrlAuthIntoServiceTemplate } from '@forms/templates/trl/trl-auth-into-service.template';
import { TrlBrakesTemplate } from '@forms/templates/trl/trl-brakes.template';
import { TrlDimensionsTemplate } from '@forms/templates/trl/trl-dimensions.template';
import { TrlPurchasers } from '@forms/templates/trl/trl-purchaser.template';
import { TrlTechRecordTemplate } from '@forms/templates/trl/trl-tech-record.template';
import { tyresTemplateTrl } from '@forms/templates/trl/trl-tyres.template';
import { TrlWeight } from '@forms/templates/trl/trl-weight.template';
import { VehicleTypes } from '@models/vehicle-tech-record.model';

// The map below initializes the array of sections that the *ngFor in tech summary component's template will iterate over.
// The order in which each section is introduced in the array will determine its order on the page when rendered.
// Sections which use custom components require a FormNode object with 'name' and 'label' properties.

export const vehicleTemplateMap = new Map<VehicleTypes, Array<FormNode>>([
  [
    VehicleTypes.PSV,
    [
      /*  1 */ TechRecordReasonForCreationSection,
      /*  2 */ PsvNotes,
      /*  3 */ PsvTechRecord,
      /*  4 */ PsvTypeApprovalTemplate,
      /*  5 */ PsvBrakesTemplate,
      /*  6 */ PsvDdaTemplate,
      /*  7 */ DocumentsTemplate,
      /*  8 */ PsvBodyTemplate,
      /*  9 */ PsvWeightsTemplate,
      /* 10 */ PsvTyresTemplate,
      /* 11 */ PsvDimensionsTemplate
    ]
  ],
  [
    VehicleTypes.HGV,
    [
      /*  1 */ TechRecordReasonForCreationSection,
      /*  2 */ NotesTemplate,
      /*  3 */ HgvTechRecord,
      /*  4 */ HgvAndTrlTypeApprovalTemplate,
      /*  5 */ ApplicantDetails,
      /*  6 */ DocumentsTemplate,
      /*  7 */ HgvAndTrlBodyTemplate,
      /*  8 */ HgvWeight,
      /*  9 */ tyresTemplateHgv,
      /* 10 */ HgvDimensionsTemplate,
      /* 11 */ PlatesTemplate
    ]
  ],
  [
    VehicleTypes.TRL,
    [
      /*  1 */ TechRecordReasonForCreationSection,
      /*  2 */ NotesTemplate,
      /*  3 */ TrlTechRecordTemplate,
      /*  4 */ HgvAndTrlTypeApprovalTemplate,
      /*  5 */ ApplicantDetails,
      /*  6 */ DocumentsTemplate,
      /*  7 */ LettersTemplate,
      /*  8 */ HgvAndTrlBodyTemplate,
      /*  9 */ TrlWeight,
      /* 10 */ tyresTemplateTrl,
      /* 11 */ TrlBrakesTemplate,
      /* 12 */ TrlPurchasers,
      /* 13 */ TrlDimensionsTemplate,
      /* 14 */ PlatesTemplate,
      /* 15 */ TrlAuthIntoServiceTemplate,
      /* 16 */ ManufacturerTemplate
    ]
  ],
  [
    VehicleTypes.SMALL_TRL,
    [/* 1 */ TechRecordReasonForCreationSection, /* 2 */ SmallTrailerTechRecord, /* 3 */ ApplicantDetails, /* 4 */ NotesTemplate, /* 5 */ Audit]
  ],
  [
    VehicleTypes.LGV,
    [/* 1 */ TechRecordReasonForCreationSection, /* 2 */ LgvTechRecord, /* 3 */ ApplicantDetails, /* 4 */ NotesTemplate, /* 5 */ Audit]
  ],
  [
    VehicleTypes.CAR,
    [/* 1 */ TechRecordReasonForCreationSection, /* 2 */ CarTechRecord, /* 3 */ ApplicantDetails, /* 4 */ NotesTemplate, /* 5 */ Audit]
  ],
  [
    VehicleTypes.MOTORCYCLE,
    [/* 1 */ TechRecordReasonForCreationSection, /* 2 */ MotorcycleTechRecord, /* 3 */ ApplicantDetails, /* 4 */ NotesTemplate, /* 5 */ Audit]
  ]
]);
