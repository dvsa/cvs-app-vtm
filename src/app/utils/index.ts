export * from './mockStore';
export * from './test-model.utils';
export * from './testing.utils';
export * from './form.utils';


