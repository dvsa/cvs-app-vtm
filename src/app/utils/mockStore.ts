import { map } from 'rxjs/operators';
import { BehaviorSubject } from 'rxjs';
import { Action } from '@ngrx/store';
import { getErrors } from '@app/store/selectors/error.selectors';
import {
  getSelectedVehicleTechRecord,
  getTechViewState,
  getVehicleTechRecordMetaData,
  getActiveVehicleTechRecord,
  selectVehicleTechRecordModelHavingStatusAll
} from '@app/store/selectors/VehicleTechRecordModel.selectors';
import {
  getSelectedVehicleTestResultModel,
  getTestViewState,
  getVehicleTestResultModel,
  selectTestTypeById
} from '@app/store/selectors/VehicleTestResultModel.selectors';
import { getPreparers, getTestStations } from '@app/store/selectors/ReferenceData.selectors';
import { getRouterParams } from '@app/store/selectors/route.selectors';

export class MockStore {
  mockSelector: BehaviorSubject<any>;

  constructor(mockSelector) {
    this.mockSelector = mockSelector;
  }

  select(selector: any) {
    switch (selector) {
      case selectVehicleTechRecordModelHavingStatusAll:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('selectVehicleTechRecordModelHavingStatusAll')
              ? value['selectVehicleTechRecordModelHavingStatusAll']
              : []
          )
        );

      case getSelectedVehicleTechRecord:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getSelectedVehicleTechRecord')
              ? value['getSelectedVehicleTechRecord']
              : {}
          )
        );

      case getActiveVehicleTechRecord:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getActiveVehicleTechRecord')
              ? value['getActiveVehicleTechRecord']
              : () => {}
          )
        );

      case getVehicleTechRecordMetaData:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getVehicleTechRecordMetaData')
              ? value['getVehicleTechRecordMetaData']
              : {}
          )
        );

      case getTechViewState:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getTechViewState') ? value['getTechViewState'] : {}
          )
        );

      case getErrors:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getErrors') ? value['getErrors'] : []
          )
        );

      case getVehicleTestResultModel:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getVehicleTestResultModel')
              ? value['getVehicleTestResultModel']
              : []
          )
        );

      case selectTestTypeById(''):
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('selectTestTypeById') ? value['selectTestTypeById'] : {}
          )
        );

      case getPreparers:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getPreparers') ? value['getPreparers'] : null
          )
        );

      case getTestStations:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getTestStations') ? value['getTestStations'] : null
          )
        );

      case getTestViewState:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getTestViewState') ? value['getTestViewState'] : {}
          )
        );

      case getSelectedVehicleTestResultModel:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getSelectedVehicleTestResultModel')
              ? value['getSelectedVehicleTestResultModel']
              : {}
          )
        );

      case getRouterParams:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getRouterParams') ? value['getRouterParams'] : {}
          )
        );

      default:
        return this.mockSelector;
    }
  }

  dispatch(action: Action) {
    return [];
  }
}
