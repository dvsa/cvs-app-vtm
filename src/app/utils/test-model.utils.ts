import { TestResultModel } from '@app/models/test-result.model';
import { TestType } from '@app/models/test.type';

export const TEST_MODEL_UTILS = {
  mockTestRecord,
  mockTestType
};

function mockTestRecord(args?: Partial<TestResultModel>): TestResultModel {
  const mock: TestResultModel = {
    reasonForCreation: '',
    testResultId: '123',
    vehicleType: 'hgv',
    systemNumber: '123',
    testTypes: [this.mockTestType()],
    testEndTimestamp: '2020-01-16T12:24:38.027Z',
    testStartTimestamp: '2020-01-16T10:24:38.027Z'
  } as TestResultModel;

  return { ...mock, ...args };
}

function mockTestType(args?: Partial<TestType>): TestType {
  const mock: TestType = {
    testTypeId: '1',
    seatbeltInstallationCheckDate: true,
    testTypeName: 'first test',
    testNumber: '123123',
    testTypeEndTimestamp: '2020-01-16T12:24:38.027Z',
    testExpiryDate: '2021-01-16T12:24:38.027Z',
    modType: {
      code: '1',
      description: 'test'
    }
  } as TestType;

  return { ...mock, ...args };
}
