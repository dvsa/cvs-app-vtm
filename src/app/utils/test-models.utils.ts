import { TestResultModel } from '../models/test-result.model';
import { TestType } from '@app/models/test.type';

/**
 * *******************SOME GROUND RULES*************************************
 *
 *
 * We use this util file essentially for unit testing to enable a clean mock object provision
 * for our unit testing. Ideally more simplified object should/can be added.
 * Amendments of returned object should be done less often or not at ALL. We use
 * partial<T> object implementation hence the required object update(s) should be
 * passed in from the specific unit test of interest.
 *
 */

export const TEST_MODEL_UTILS = {
    mockTestResult
  };


function mockTestResult(args?: Partial<TestResultModel>): TestResultModel {
    const mock: TestResultModel = {
      testEndTimestamp: '2020-01-16T12:24:38.027Z',
      testStartTimestamp: '2020-01-16T10:24:38.027Z',
      testTypes: [mockTestType()]
    } as TestResultModel;
    return { ...mock, ...args };
  }

  function mockTestType(args?: Partial<TestType>): TestType {
    const mock: TestType = {
      testTypeName: 'first test',
      testNumber: '123123',
      testTypeEndTimestamp: '2020-01-16T12:24:38.027Z',
      testExpiryDate: '2021-01-16T12:24:38.027Z'
    } as TestType;
    return { ...mock, ...args };
  }
