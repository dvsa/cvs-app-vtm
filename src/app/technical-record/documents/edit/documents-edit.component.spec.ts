import {DebugElement} from '@angular/core';
import {async, ComponentFixture, TestBed} from '@angular/core/testing';

import {
  AbstractControl,
  FormsModule,
  ReactiveFormsModule,
  FormGroupDirective,
  FormGroup
} from '@angular/forms';

import {DocumentsEditComponent} from './documents-edit.component';
import { TESTING_UTILS } from '@app/utils';


describe('DocumentsEditComponent component', () => {
  let component: DocumentsEditComponent;
  let fixture: ComponentFixture<DocumentsEditComponent>;
  let de: DebugElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [FormsModule, ReactiveFormsModule],
      declarations: [DocumentsEditComponent],
      providers: [
        FormGroupDirective,
        {
          provide: FormGroupDirective,
          useValue: TESTING_UTILS.mockFormGroupDirective({
            techRecord: new FormGroup({
            }) as AbstractControl
          })
        }
      ]
    })
    .compileComponents();

  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(DocumentsEditComponent);
    component = fixture.componentInstance;
    component.microfilmDetails = TESTING_UTILS.mockMicrofilm();
    de = fixture.debugElement;

    fixture.detectChanges();
  });


  it('should render with the given properties', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

});
