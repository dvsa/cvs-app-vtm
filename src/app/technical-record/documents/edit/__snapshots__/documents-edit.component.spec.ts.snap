// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentsEditComponent component should render with the given properties 1`] = `
<vtm-documents-edit
  fb={[Function FormBuilder]}
  microfilmDetails={[Function Object]}
  microfilmDocumentTypeOptions={[Function Array]}
  parent={[Function FormGroupDirective]}
  techRecordFg={[Function FormGroup]}
>
  <div
    class="govuk-grid-row"
  >
    <div
      class="govuk-grid-column-three-quarters"
    >
      <fieldset
        class="govuk-fieldset"
      >
        <form
          class="ng-untouched ng-pristine ng-valid"
          ng-reflect-form="[object Object]"
          novalidate=""
        >
          
          <div
            class="govuk-form-group"
          >
            <label
              class="govuk-label govuk-!-font-weight-bold"
              for="microfilmDocumentType"
            >
               Microfilm document type (optional) 
            </label>
            <select
              class="govuk-select ng-untouched ng-pristine ng-valid"
              formcontrolname="microfilmDocumentType"
              id="microfilmDocumentType"
              ng-reflect-name="microfilmDocumentType"
            >
              
              <option
                ng-reflect-ng-value="PSV Miscellaneous"
                value="0: PSV Miscellaneous"
              >
                 PSV Miscellaneous 
              </option>
              <option
                ng-reflect-ng-value="AAT - Trailer Annual Test"
                value="1: AAT - Trailer Annual Test"
              >
                 AAT - Trailer Annual Test 
              </option>
              <option
                ng-reflect-ng-value="AIV - HGV International App"
                value="2: AIV - HGV International App"
              >
                 AIV - HGV International App 
              </option>
              <option
                ng-reflect-ng-value="COIF Modification"
                value="3: COIF Modification"
              >
                 COIF Modification 
              </option>
              <option
                ng-reflect-ng-value="Trailer COC + Int Plate"
                value="4: Trailer COC + Int Plate"
              >
                 Trailer COC + Int Plate 
              </option>
              <option
                ng-reflect-ng-value="RCT - Trailer Test Cert paid"
                value="5: RCT - Trailer Test Cert paid"
              >
                 RCT - Trailer Test Cert paid 
              </option>
              <option
                ng-reflect-ng-value="HGV COC + Int Plate"
                value="6: HGV COC + Int Plate"
              >
                 HGV COC + Int Plate 
              </option>
              <option
                ng-reflect-ng-value="PSV Carry/Auth, OMO Report"
                value="7: PSV Carry/Auth, OMO Report"
              >
                 PSV Carry/Auth, OMO Report 
              </option>
              <option
                ng-reflect-ng-value="AIT - Trailer International Ap"
                value="8: AIT - Trailer International App"
              >
                 AIT - Trailer International App 
              </option>
              <option
                ng-reflect-ng-value="IPV - HGV EEC Plate/Cert"
                value="9: IPV - HGV EEC Plate/Cert"
              >
                 IPV - HGV EEC Plate/Cert 
              </option>
              <option
                ng-reflect-ng-value="XCV - HGV Test Cert free"
                value="10: XCV - HGV Test Cert free"
              >
                 XCV - HGV Test Cert free 
              </option>
              <option
                ng-reflect-ng-value="AAV - HGV Annual Test"
                value="11: AAV - HGV Annual Test"
              >
                 AAV - HGV Annual Test 
              </option>
              <option
                ng-reflect-ng-value="COIF Master"
                value="12: COIF Master"
              >
                 COIF Master 
              </option>
              <option
                ng-reflect-ng-value="Tempo 100 Sp Ord"
                value="13: Tempo 100 Sp Ord"
              >
                 Tempo 100 Sp Ord 
              </option>
              <option
                ng-reflect-ng-value="Deleted"
                value="14: Deleted"
              >
                 Deleted 
              </option>
              <option
                ng-reflect-ng-value="PSV N/ALT"
                value="15: PSV N/ALT"
              >
                 PSV N/ALT 
              </option>
              <option
                ng-reflect-ng-value="XPT - Tr Plating Cert paid"
                value="16: XPT - Tr Plating Cert paid"
              >
                 XPT - Tr Plating Cert paid 
              </option>
              <option
                ng-reflect-ng-value="FFV - HGV First Test"
                value="17: FFV - HGV First Test"
              >
                 FFV - HGV First Test 
              </option>
              <option
                ng-reflect-ng-value="Repl Vitesse 100"
                value="18: Repl Vitesse 100"
              >
                 Repl Vitesse 100 
              </option>
              <option
                ng-reflect-ng-value="TCV - HGV Test Cert"
                value="19: TCV - HGV Test Cert"
              >
                 TCV - HGV Test Cert 
              </option>
              <option
                ng-reflect-ng-value="ZZZ - Miscellaneous"
                value="20: ZZZ - Miscellaneous"
              >
                 ZZZ - Miscellaneous 
              </option>
              <option
                ng-reflect-ng-value="Test Certificate"
                value="21: Test Certificate"
              >
                 Test Certificate 
              </option>
              <option
                ng-reflect-ng-value="XCT - Trailer Test Cert free"
                value="22: XCT - Trailer Test Cert free"
              >
                 XCT - Trailer Test Cert free 
              </option>
              <option
                ng-reflect-ng-value="C52 - COC and VTG52A"
                value="23: C52 - COC and VTG52A"
              >
                 C52 - COC and VTG52A 
              </option>
              <option
                ng-reflect-ng-value="Tempo 100 Report"
                value="24: Tempo 100 Report"
              >
                 Tempo 100 Report 
              </option>
              <option
                ng-reflect-ng-value="Main File Amendment"
                value="25: Main File Amendment"
              >
                 Main File Amendment 
              </option>
              <option
                ng-reflect-ng-value="PSV Doc"
                value="26: PSV Doc"
              >
                 PSV Doc 
              </option>
              <option
                ng-reflect-ng-value="PSV COC"
                value="27: PSV COC"
              >
                 PSV COC 
              </option>
              <option
                ng-reflect-ng-value="PSV Repl COC"
                value="28: PSV Repl COC"
              >
                 PSV Repl COC 
              </option>
              <option
                ng-reflect-ng-value="TAV - COC"
                value="29: TAV - COC"
              >
                 TAV - COC 
              </option>
              <option
                ng-reflect-ng-value="NPT - Trailer Alteration"
                value="30: NPT - Trailer Alteration"
              >
                 NPT - Trailer Alteration 
              </option>
              <option
                ng-reflect-ng-value="OMO Certificate"
                value="31: OMO Certificate"
              >
                 OMO Certificate 
              </option>
              <option
                ng-reflect-ng-value="PSV Repl COIF"
                value="32: PSV Repl COIF"
              >
                 PSV Repl COIF 
              </option>
              <option
                ng-reflect-ng-value="PSV Repl COF"
                value="33: PSV Repl COF"
              >
                 PSV Repl COF 
              </option>
              <option
                ng-reflect-ng-value="COIF Application"
                value="34: COIF Application"
              >
                 COIF Application 
              </option>
              <option
                ng-reflect-ng-value="XPV - HGV Plating Cert Free"
                value="35: XPV - HGV Plating Cert Free"
              >
                 XPV - HGV Plating Cert Free 
              </option>
              <option
                ng-reflect-ng-value="TCT - Trailer Test Cert"
                value="36: TCT - Trailer Test Cert"
              >
                 TCT - Trailer Test Cert 
              </option>
              <option
                ng-reflect-ng-value="Tempo 100 App"
                value="37: Tempo 100 App"
              >
                 Tempo 100 App 
              </option>
              <option
                ng-reflect-ng-value="PSV Decision on N/ALT"
                value="38: PSV Decision on N/ALT"
              >
                 PSV Decision on N/ALT 
              </option>
              <option
                ng-reflect-ng-value="Special Order PSV"
                value="39: Special Order PSV"
              >
                 Special Order PSV 
              </option>
              <option
                ng-reflect-ng-value="NPV - HGV Alteration"
                value="40: NPV - HGV Alteration"
              >
                 NPV - HGV Alteration 
              </option>
              <option
                ng-reflect-ng-value="No Description Found"
                value="41: No Description Found"
              >
                 No Description Found 
              </option>
              <option
                ng-reflect-ng-value="Vitesse 100 Sp Ord"
                value="42: Vitesse 100 Sp Ord"
              >
                 Vitesse 100 Sp Ord 
              </option>
              <option
                ng-reflect-ng-value="Brake Test Details"
                value="43: Brake Test Details"
              >
                 Brake Test Details 
              </option>
              <option
                ng-reflect-ng-value="COIF Productional"
                value="44: COIF Productional"
              >
                 COIF Productional 
              </option>
              <option
                ng-reflect-ng-value="RDT - Test Disc Paid"
                value="45: RDT - Test Disc Paid"
              >
                 RDT - Test Disc Paid 
              </option>
              <option
                ng-reflect-ng-value="RCV - HGV Test Cert"
                value="46: RCV - HGV Test Cert"
              >
                 RCV - HGV Test Cert 
              </option>
              <option
                ng-reflect-ng-value="FFT - Trailer First Test"
                value="47: FFT - Trailer First Test"
              >
                 FFT - Trailer First Test 
              </option>
              <option
                ng-reflect-ng-value="IPT - Trailer EEC Plate/Cert"
                value="48: IPT - Trailer EEC Plate/Cert"
              >
                 IPT - Trailer EEC Plate/Cert 
              </option>
              <option
                ng-reflect-ng-value="XDT - Test Disc Free"
                value="49: XDT - Test Disc Free"
              >
                 XDT - Test Disc Free 
              </option>
              <option
                ng-reflect-ng-value="PRV - HGV Plating Cert paid"
                value="50: PRV - HGV Plating Cert paid"
              >
                 PRV - HGV Plating Cert paid 
              </option>
              <option
                ng-reflect-ng-value="COF Cert, PRT - Tr Plating Cer"
                value="51: COF Cert, PRT - Tr Plating Cert paid"
              >
                 COF Cert, PRT - Tr Plating Cert paid 
              </option>
              <option
                ng-reflect-ng-value="Tempo 100 Permit"
                value="52: Tempo 100 Permit"
              >
                 Tempo 100 Permit 
              </option>
            </select>
          </div>
          <div
            class="govuk-form-group"
          >
            <label
              class="govuk-label govuk-!-font-weight-bold"
              for="microfilmRollNumber"
            >
               Microfilm roll number (optional) 
            </label>
            <input
              class="govuk-input govuk-input--width-10 ng-untouched ng-pristine ng-valid"
              formcontrolname="microfilmRollNumber"
              id="microfilmRollNumber"
              maxlength="5"
              ng-reflect-maxlength="5"
              ng-reflect-name="microfilmRollNumber"
              type="text"
            />
          </div>
          <div
            class="govuk-form-group"
          >
            <label
              class="govuk-label govuk-!-font-weight-bold"
              for="microfilmSerialNumber"
            >
               Microfilm serial number (optional) 
            </label>
            <input
              class="govuk-input govuk-input--width-5 ng-untouched ng-pristine ng-valid"
              formcontrolname="microfilmSerialNumber"
              id="microfilmSerialNumber"
              maxlength="4"
              ng-reflect-maxlength="4"
              ng-reflect-name="microfilmSerialNumber"
              type="text"
            />
          </div>
        </form>
      </fieldset>
    </div>
  </div>
</vtm-documents-edit>
`;
